esmvalcore/_data_finder.py

```diff
@@ -37,15 +37,28 @@ def get_start_end_year(filename):
     stem = Path(filename).stem
     start_year = end_year = None
 
+    # We use (?:) in order not to capture pairs of digits, but
+    # directly whole date
+    date_pattern = r"((?:[0-9]{2}){2,6})"
```

Comment:
I like it! Perhaps add a small comment explaining the logic of always dealing with pairs of digits instead of individual digits?

Reply:
Done